src/Components/Components/src/Routing/QueryParameterNameComparer.cs

```diff
@@ -0,0 +1,26 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Components.Routing
+{
+    internal class QueryParameterNameComparer : IComparer<ReadOnlyMemory<char>>, IEqualityComparer<ReadOnlyMemory<char>>
+    {
+        public static readonly QueryParameterNameComparer Instance = new();
+
+        public int Compare(ReadOnlyMemory<char> x, ReadOnlyMemory<char> y)
+            => x.Span.CompareTo(y.Span, StringComparison.OrdinalIgnoreCase);
+
+        public bool Equals(ReadOnlyMemory<char> x, ReadOnlyMemory<char> y)
+            => x.Span.Equals(y.Span, StringComparison.OrdinalIgnoreCase);
+
+        public int GetHashCode([DisallowNull] ReadOnlyMemory<char> obj)
```

Comment:
Weird you had to do `DisallowNull`. Is that part of the signature?

Reply:
This implementation was pulled out of `QueryParameterValueSupplier.cs`, but also I implemented my own comparer before doing that, and iirc it auto generated the `DisallowNull` attribute. Found this discussion about it: https://github.com/dotnet/runtime/issues/30998.